File: src/C3.Catalogue/Decks/DeckDetails.vb
Namespace Decks

    Public NotInheritable Class DeckDetails

        Public Sub New(
                manufacturer As String,
                model As String,
                year As Integer,
                condition As Integer,
                type1 As Boolean,
                type2 As Boolean,
                type3 As Boolean,
                type4 As Boolean,
                hx As Boolean,
                mpx As Boolean,
                dolbyB As Boolean,
                dolbyC As Boolean,
                dolbyS As Boolean,
                dbx1 As Boolean,
                dbx2 As Boolean,
                stereo As Boolean,
                programSearch As Boolean,
                reverse As Boolean,
                calibration As Boolean,
                azimuth As Boolean,
                dubbingSlow As Boolean,
                dubbingFast As Boolean,
                frequencyLow As Integer,
                frequencyHigh As Integer,
                signalRatio As Integer,
                signalRatioNoiseReduction As String,
                wowFlutter As Decimal,
                distortion As Decimal,
                heads As Integer,
                wells As Integer,
                speedSlow As Boolean,
                speedNormal As Boolean,
                speedFast As Boolean,
                notes As String)

            Me.Manufacturer = manufacturer
            Me.Model = model
            Me.Year = year
            Me.Condition = condition
            Me.Type1 = type1
            Me.Type2 = type2
            Me.Type3 = type3
            Me.Type4 = type4
            Me.Hx = hx
            Me.Mpx = mpx
            Me.DolbyB = dolbyB
            Me.DolbyC = dolbyC
            Me.DolbyS = dolbyS
            Me.Dbx1 = dbx1
            Me.Dbx2 = dbx2
            Me.Stereo = stereo
            Me.ProgramSearch = programSearch
            Me.Reverse = reverse
            Me.Calibration = calibration
            Me.Azimuth = azimuth
            Me.DubbingSlow = dubbingSlow
            Me.DubbingFast = dubbingFast
            Me.FrequencyLow = frequencyLow
            Me.FrequencyHigh = frequencyHigh
            Me.SignalRatio = signalRatio
            Me.SignalRatioNoiseReduction = signalRatioNoiseReduction
            Me.WowFlutter = wowFlutter
            Me.Distortion = distortion
            Me.Heads = heads
            Me.Wells = wells
            Me.SpeedSlow = speedSlow
            Me.SpeedNormal = speedNormal
            Me.SpeedFast = speedFast
            Me.Notes = notes
        End Sub

        Public ReadOnly Property Manufacturer As String
        Public ReadOnly Property Model As String
        Public ReadOnly Property Year As Integer
        Public ReadOnly Property Condition As Integer
        Public ReadOnly Property Type1 As Boolean
        Public ReadOnly Property Type2 As Boolean
        Public ReadOnly Property Type3 As Boolean
        Public ReadOnly Property Type4 As Boolean
        Public ReadOnly Property Hx As Boolean
        Public ReadOnly Property Mpx As Boolean
        Public ReadOnly Property DolbyB As Boolean
        Public ReadOnly Property DolbyC As Boolean
        Public ReadOnly Property DolbyS As Boolean
        Public ReadOnly Property Dbx1 As Boolean
        Public ReadOnly Property Dbx2 As Boolean
        Public ReadOnly Property Stereo As Boolean
        Public ReadOnly Property ProgramSearch As Boolean
        Public ReadOnly Property Reverse As Boolean
        Public ReadOnly Property Calibration As Boolean
        Public ReadOnly Property Azimuth As Boolean
        Public ReadOnly Property DubbingSlow As Boolean
        Public ReadOnly Property DubbingFast As Boolean
        Public ReadOnly Property FrequencyLow As Integer
        Public ReadOnly Property FrequencyHigh As Integer
        Public ReadOnly Property SignalRatio As Integer
        Public ReadOnly Property SignalRatioNoiseReduction As String
        Public ReadOnly Property WowFlutter As Decimal
        Public ReadOnly Property Distortion As Decimal
        Public ReadOnly Property Heads As Integer
        Public ReadOnly Property Wells As Integer
        Public ReadOnly Property SpeedSlow As Boolean
        Public ReadOnly Property SpeedNormal As Boolean
        Public ReadOnly Property SpeedFast As Boolean
        Public ReadOnly Property Notes As String

    End Class

End Namespace

File: src/C3.Catalogue/Decks/DeckDetails.cs
namespace C3.Catalogue.Decks
{
    public sealed class DeckDetails
    {
        public DeckDetails(
            string manufacturer,
            string model,
            int year,
            int condition,
            bool type1,
            bool type2,
            bool type3,
            bool type4,
            bool hx,
            bool mpx,
            bool dolbyB,
            bool dolbyC,
            bool dolbyS,
            bool dbx1,
            bool dbx2,
            bool stereo,
            bool programSearch,
            bool reverse,
            bool calibration,
            bool azimuth,
            bool dubbingSlow,
            bool dubbingFast,
            int frequencyLow,
            int frequencyHigh,
            int signalRatio,
            string signalRatioNoiseReduction,
            decimal wowFlutter,
            decimal distortion,
            int heads,
            int wells,
            bool speedSlow,
            bool speedNormal,
            bool speedFast,
            string notes)
        {
            Manufacturer = manufacturer;
            Model = model;
            Year = year;
            Condition = condition;
            Type1 = type1;
            Type2 = type2;
            Type3 = type3;
            Type4 = type4;
            Hx = hx;
            Mpx = mpx;
            DolbyB = dolbyB;
            DolbyC = dolbyC;
            DolbyS = dolbyS;
            Dbx1 = dbx1;
            Dbx2 = dbx2;
            Stereo = stereo;
            ProgramSearch = programSearch;
            Reverse = reverse;
            Calibration = calibration;
            Azimuth = azimuth;
            DubbingSlow = dubbingSlow;
            DubbingFast = dubbingFast;
            FrequencyLow = frequencyLow;
            FrequencyHigh = frequencyHigh;
            SignalRatio = signalRatio;
            SignalRatioNoiseReduction = signalRatioNoiseReduction;
            WowFlutter = wowFlutter;
            Distortion = distortion;
            Heads = heads;
            Wells = wells;
            SpeedSlow = speedSlow;
            SpeedNormal = speedNormal;
            SpeedFast = speedFast;
            Notes = notes;
        }

        public string Manufacturer { get; }
        public string Model { get; }
        public int Year { get; }
        public int Condition { get; }
        public bool Type1 { get; }
        public bool Type2 { get; }
        public bool Type3 { get; }
        public bool Type4 { get; }
        public bool Hx { get; }
        public bool Mpx { get; }
        public bool DolbyB { get; }
        public bool DolbyC { get; }
        public bool DolbyS { get; }
        public bool Dbx1 { get; }
        public bool Dbx2 { get; }
        public bool Stereo { get; }
        public bool ProgramSearch { get; }
        public bool Reverse { get; }
        public bool Calibration { get; }
        public bool Azimuth { get; }
        public bool DubbingSlow { get; }
        public bool DubbingFast { get; }
        public int FrequencyLow { get; }
        public int FrequencyHigh { get; }
        public int SignalRatio { get; }
        public string SignalRatioNoiseReduction { get; }
        public decimal WowFlutter { get; }
        public decimal Distortion { get; }
        public int Heads { get; }
        public int Wells { get; }
        public bool SpeedSlow { get; }
        public bool SpeedNormal { get; }
        public bool SpeedFast { get; }
        public string Notes { get; }
    }
}
